Allow users to remove themselves from groups

ryanlerch opened the pull request:
Adds an ACI that allows users to remove themselves from a group.

ACI written by @tiran in
https://github.com/fedora-infra/securitas/issues/108#issuecomment-585718220

Related: https://github.com/fedora-infra/securitas/issues/108

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

tiran commented on updates/99-fas.update:
```diff
@@ -20,6 +20,11 @@ add:aci: (targetattr = "fasTimezone || fasLocale || fasIRCNick || fasGPGKeyId ||
 
+
+# self-service permissions to allow users to remove themselves from a group
+dn: $SUFFIX
```
Group related ACIs should be added to the group container ``cn=groups,cn=accounts,$SUFFIX``.

tiran commented on updates/99-fas.update:
```diff
@@ -20,6 +20,11 @@ add:aci: (targetattr = "fasTimezone || fasLocale || fasIRCNick || fasGPGKeyId ||
+
+# self-service permissions to allow users to remove themselves from a group
+dn: $SUFFIX
+add:aci: (targetattr = "member")(targetfilter = "(objectclass=ipaUserGroup)")(version 3.0; acl "Allow member to remove themselves"; allow (selfwrite) userattr = "member#USERDN";)
```
My ACI had a typo:
> Allow member**s** to remove themselves

tiran commented on updates/99-fas.update:
```diff
@@ -20,6 +20,11 @@ add:aci: (targetattr = "fasTimezone || fasLocale || fasIRCNick || fasGPGKeyId ||
 add:aci: (targetattr = "mail")(version 3.0;acl "selfservice:Users can manage their own email address";allow (write) userdn = "ldap:///self";)
 
+
+# self-service permissions to allow users to remove themselves from a group
```
By convention self-service permissions are permissions with ``selfservice:`` prefix.

ryanlerch commented:
Thanks for the feedback! Updated the commit with the requested changes.

I also updated the comment in 99-fas.update, to refer to "members" rather than "users".
